docs(observability_pipelines): add WebSocket source page (#37197)
DataDog/documentation: content/en/observability_pipelines/sources/websocket.md

---
title: WebSocket Source

> disable_toc: false

Reviewer: ```suggestion
disable_toc: false
description: Learn how to connect the Observability Pipelines Worker to a WebSocket and ingest the messages as logs.

```

products:
- name: Logs
  icon: logs
  url: /observability_pipelines/configuration/?tab=logs#pipeline-types
---

{{< product-availability >}}

{{< callout url="#" btn_hidden="true" header="Join the Preview!">}}
The WebSocket source is in Preview. Contact your account manager to request access.
{{< /callout >}}

## Overview

> Use Observability Pipelines' WebSocket source to connect the Observability Pipelines Worker as a client to an upstream `ws://` or `wss://` endpoint and ingest the messages it receives as logs. Use this source when your upstream system streams data over a persistent WebSocket connection rather than exposing it through a request-based API.

Reviewer: "...rather than exposing it..." What's getting exposed, the data?

```suggestion
Use Observability Pipelines' WebSocket source to connect the Observability Pipelines Worker as a client to an upstream `ws://` or `wss://` endpoint. The Worker ingests the messages as logs. Use this source when your upstream system sends data over a persistent WebSocket connection rather than exposing the data through a request-based API.
```

## Prerequisites

> {{% observability_pipelines/prerequisites/websocket %}}

Reviewer: I don't think this needs to be a shortcode. Is it getting used elsewhere? If not, the content can just be added to this file.

## Setup

Set up this source when you [set up a pipeline][1]. You can set up a pipeline in the [UI][3], using the [API][4], or with [Terraform][5]. The instructions in this section are for setting up the source in the UI.

<div class="alert alert-danger">For Secrets Management: Only enter the identifiers for the WebSocket URI and, if applicable, your authorization strategy secrets and TLS key pass. Do <b>not</b> enter the actual values.</div>

{{% observability_pipelines/secrets_env_var_note %}}

After you select the WebSocket source in the pipeline UI:

> 1. Enter the identifier for your WebSocket URI. This is the name of the environment variable that holds the `ws://` or `wss://` endpoint that the Worker connects to. If you leave it blank, the [default](#secret-defaults) is used.

Reviewer: ```suggestion
1. Enter the identifier for your WebSocket URI, which references the `ws://` or `wss://` endpoint that the Worker connects to. If you leave it blank, the [default](#secret-defaults) is used.
```

1. Select your authorization strategy. If you selected:
   - **None**: No authentication is sent with the connection request. Use this when the endpoint does not require credentials.
   - **Basic**:
      - Enter the identifier for your WebSocket username. If you leave it blank, the [default](#secret-defaults) is used.

>       - Enter the identifier for your WebSocket password. If you leave it blank, the [default](#secret-defaults) is used.

Reviewer: ```suggestion
   - **Basic**: Enter the identifiers for your WebSocket username and password. If you leave it blank, the [default](#secret-defaults) is used.
```

   - **Bearer**: Enter the identifier for your bearer token. The token is sent in the `Authorization` header as `Bearer <token>`. If you leave it blank, the [default](#secret-defaults) is used.
   - **Custom**: Enter the identifier for your custom `Authorization` header value. The Worker sends this value as the `Authorization` header exactly as provided. Use this when the endpoint expects an authentication scheme other than basic or bearer. If you leave it blank, the [default](#secret-defaults) is used.
1. In the **Decoding** dropdown menu, select the decoder to apply to incoming messages. Messages received from the endpoint must be in the selected format.

   | Decoding | Description |
   | -------- | ----------- |

>    | `bytes` | Pass each message through as raw bytes. The raw message is stored in the `message` field. |

Reviewer: ```suggestion
   | `bytes` | Send each message as raw bytes. The raw message is stored in the `message` field. |
```

   | `gelf`  | Decode each message as a Graylog Extended Log Format (GELF) event. |
   | `json`  | Decode each message as a JSON object. |
   | `syslog`| Decode each message as a Syslog-formatted event. |

>    If your messages do not match one of the formats above, select `bytes` to ingest the raw message in the `message` field, then transform it with the [Custom Processor][6] using Vector Remap Language (VRL).

Reviewer: ```suggestion
   If your messages do not match one of the formats above, select `bytes` to ingest the raw message in the `message` field, then transform it with a [Custom Processor][6].
```

### Optional TLS settings

For `wss://` endpoints, configure TLS so that data is encrypted in transit. The WebSocket source supports two TLS modes:

#### Enable TLS

> Toggle the switch to **Enable TLS**. Because the Worker connects to the endpoint as a client, no certificate fields are required: enabling TLS encrypts the connection, and the endpoint must present a certificate signed by a trusted Certificate Authority (CA). Observability Pipelines does not accept self-signed certificates by default.

Reviewer: I'm not sure if we need to mention:

> Observability Pipelines does not accept self-signed certificates by default.

```suggestion
Toggle the switch to **Enable TLS**. Because the Worker connects to the endpoint as a client, no certificate fields are required. Enabling TLS encrypts the connection and the endpoint must present a certificate signed by a trusted Certificate Authority (CA).
```

---
title: WebSocket Source
disable_toc: false
products:
- name: Logs
  icon: logs
  url: /observability_pipelines/configuration/?tab=logs#pipeline-types
---

{{< product-availability >}}

## Overview

Use Observability Pipelines' WebSocket source to connect the Observability Pipelines Worker as a client to an upstream `ws://` or `wss://` endpoint and ingest the messages it receives as logs. Use this source when your upstream system streams data over a persistent WebSocket connection rather than exposing it through a request-based API.

## Prerequisites

{{% observability_pipelines/prerequisites/websocket %}}

## Setup

> Set up this source when you [set up a pipeline][1]. You can set up a pipeline in the [UI][3], using the [API][4], or with [Terraform][5]. The instructions in this section are for setting up the source in the UI.

Author: Keeping this as-is for consistency. This sentence is the shared boilerplate used by every Observability Pipelines source page (e.g. `http_client.md` line 22 is identical) — it describes the general pipeline-setup options, and links [4]/[5] point to the generic API reference and the Terraform provider registry, not WebSocket-specific instructions. The page already scopes itself with "The instructions in this section are for setting up the source in the UI."

WebSocket API and Terraform support are shipping in parallel as part of this same rollout, so they'll be live by the time this page publishes. Special-casing only this page to drop the API/Terraform links would make it diverge from all sibling source pages and would be incorrect once those land.
